packages/vsce/src/utils/profileManagement.ts

```diff
@@ -52,273 +53,292 @@ export class ProfileManagement {
     return mProfileInfo;
   }
 
-  /**
-   * Makes axios GET request with path and axios config object given
-   * @param path
-   * @param config
-   * @returns
-   */
-  public static async makeRequest(path: string, config: AxiosRequestConfig) {
-    const response = await axios.get(path, config);
-    return response;
-  }
-
   public static cmciResponseXml2Json(data: string) {
     return JSON.parse(xml2json(data, { compact: true, spaces: 4 }));
   }
 
-  /**
-   * Populates the info
-   * @param profile
-   * @returns Array of type InfoLoaded
-   */
-  public static async getPlexInfo(profile: imperative.IProfileLoaded): Promise<InfoLoaded[]> {
-    const config: AxiosRequestConfig = {
-      baseURL: `${profile.profile.protocol}://${profile.profile.host}:${profile.profile.port}/CICSSystemManagement`,
-      auth: {
-        username: profile.profile.user,
-        password: profile.profile.password,
-      },
-    };
+  public static getSessionFromProfile(profile: imperative.IProfile): Session {
+    return new Session({
+      protocol: profile.protocol,
+      hostname: profile.host,
+      port: profile.port,
+      type: "basic",
+      user: profile.user,
+      password: profile.password,
+      rejectUnauthorized: 'rejectUnauthorized' in profile ? profile.rejectUnauthorized : true,
+    });
+  }
+
+  public static async regionIsGroup(session: Session, profile: imperative.IProfile): Promise<boolean> {
+
+    let checkIfSystemGroup: ICMCIApiResponse;
+    try {
+      checkIfSystemGroup = await getResource(session, {
+        name: "CICSRegionGroup",
+        cicsPlex: profile.cicsPlex,
+        regionName: profile.regionName,
+        criteria: `GROUP=${profile.regionName}`,
+      });
+    } catch (error) {
+      if (error instanceof imperative.ImperativeError) {
+        if (!error.mDetails.msg.toUpperCase().includes("NODATA")) {
+          throw error;
+        }
+      }
+    }
+
+    return checkIfSystemGroup?.response.resultsummary.recordcount !== "0";
+  }
```

Comment:
If we get an error (like NODATA), checkIfSystemGroup is still undefined, which means it isn't "0", which means we return true ... which is wrong.

Making the flow clearer and ensuring we set group to false if checkIfSystemGroup is undefined will give the right result.

```suggestion
  public static async regionIsGroup(session: Session, profile: imperative.IProfile): Promise<boolean> {
    let isGroup = false;

    try {
      const checkIfSystemGroup: ICMCIApiResponse = await getResource(session, {
        name: "CICSRegionGroup",
        cicsPlex: profile.cicsPlex,
        regionName: profile.regionName,
        criteria: `GROUP=${profile.regionName}`,
      });

      if (checkIfSystemGroup && checkIfSystemGroup.response.resultsummary.recordcount !== "0") {
        isGroup = true;
      }
    } catch (error) {
      if (error instanceof imperative.ImperativeError) {
        if (!error.mDetails.msg.toUpperCase().includes("NODATA")) {
          throw error;
        }
      }
    }

    return isGroup;
  }
```